analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/ExceptionCandidate.cs

```diff
@@ -31,28 +31,34 @@ internal class ExceptionCandidate
         public ExceptionCandidate(Compilation compilation) =>
             typeCatalog = new TypeCatalog(compilation);
 
+        public static bool IsMonitorExit(IInvocationOperationWrapper invocation) =>
+            invocation.TargetMethod.Is(KnownType.System_Threading_Monitor, "Exit");
+
+        public static bool IsLockRelease(IInvocationOperationWrapper invocation) =>
+            invocation.TargetMethod.IsAny(KnownType.System_Threading_ReaderWriterLock, "ReleaseLock", "ReleaseReaderLock", "ReleaseWriterLock")
+            || invocation.TargetMethod.IsAny(KnownType.System_Threading_ReaderWriterLockSlim, "ExitReadLock", "ExitWriteLock", "ExitUpgradeableReadLock")
+            || invocation.TargetMethod.Is(KnownType.System_Threading_Mutex, "ReleaseMutex")
+            || invocation.TargetMethod.Is(KnownType.System_Threading_SpinLock, "Exit");
+
```

Comment:
I think that in terms of responsibilities these do not belong here. Why would `LocksReleasedAllPathsBase` know about `ExceptionCandidate` that is a SEE detail?
What do you think about extracting `IInvocationOperationWrapperExtensions` and move these functions there as extension methods?

Reply:
There's a functional dependency here. All of these has to be considered on both sides. If I mechanically extract them to 3rd location, the dependency will become hidden.

Reply:
But this dependency also looks indirect and hidden. I think a comment would provide more details instead.